jwst/stpipe/core.py

```diff
@@ -92,6 +99,12 @@ def finalize_result(self, result, reference_files_used):
                 if self.parent is None:
                     log.info(f"Results used CRDS context: {result.meta.ref_file.crds.context_used}")
 
+            if hasattr(result, 'cal_logs'):
+                tmpdict = result.cal_logs.instance
+            else:
+                tmpdict = dict()
+            tmpdict[self.class_alias] = '\n'.join(self._log_records)
+            result.cal_logs = tmpdict
```

Comment:
```suggestion
            if not hasattr(result, 'cal_logs'):
                result.cal_logs = {}
            setattr(result.cal_logs, self.class_alias, self._log_records)
```
Some code golf and switchingg the `cal_log` to a list of strings.